sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -57,23 +58,30 @@ async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
 
         public async Task<ServiceBusTriggerMetrics> GetMetricsAsync()
         {
-            ServiceBusReceivedMessage message = null;
+            ServiceBusReceivedMessage activeMessage = null;
             string entityName = _serviceBusEntityType == ServiceBusEntityType.Queue ? "queue" : "topic";
 
             try
             {
-                // Peek the first message in the queue without removing it from the queue
-                // PeekAsync remembers the sequence number of the last message, so the second call returns the second message instead of the first one
-                // Use PeekBySequenceNumberAsync with fromSequenceNumber = 0 to always get the first available message
-                message = await _receiver.Value.PeekMessageAsync(fromSequenceNumber: 0).ConfigureAwait(false);
+                var receivedMessages = await _receiver.Value.PeekMessagesAsync(10, fromSequenceNumber: lastSequenceNumber).ConfigureAwait(false);
```

Comment:
Also a small nit - let's call the local peekedMessages rather than receivedMessages since they're not actually being received/dequeued?

Reply:
Sure will do